packages/block-editor/src/components/block-draggable/index.js

```diff
@@ -30,6 +64,14 @@ const BlockDraggable = ( { children, clientIds, cloneClassname } ) => {
 		[ clientIds ]
 	);
 	const isDragging = useRef( false );
+
+	// @todo - do this for horizontal scroll
+	const dragStartY = useRef( null );
+	const velocityY = useRef( null );
+	const scrollParentY = useRef( null );
+
+	const scrollEditorInterval = useRef( null );
```

Comment:
Do you think there's away to extract the logic into a dedicated hook as it's kind of obscures the code of the component. Would something like that (or close) work?

```
const [ onDragStart, onDragOver, onDragEnd ] = useScrollOnDrag()
```

Reply:
Maybe. I don't have a ton of bandwidth to refine this so you should feel free to mess around with it and propose a better-factored alternative.